### PR TITLE
23 sap quickstart basic checkov implementation

### PR DESCRIPTION
**Pull request by eduardodfmex**
This pull request introduces several changes to improve the integration of Microsoft's Power Platform and Azure services with SAP Systems. The most important changes include the addition of new terms in the `.vscode/settings.json` file, modifications to the `README.md` file, changes in the `bootstrap/bootstrap.sh` file, and the removal of a lock file. Additionally, a new `README.md` file was added in the `quickstarts/301-sap-gateway` directory, along with changes to the `main.tf`, `outputs.tf`, and `variables.tf` files.

Additions to the `.vscode/settings.json` file:

* Added new terms "quickstartergen" and "quickstartgen" to the list of terms.

Changes to the `README.md` file:

* Modified the "Tooling" section to change the note format.
* Added a new note in the "Quickstart Examples" section, added a new quickstart example, and added a link to the PowerPlatform provider documentation.

Changes to the `bootstrap/bootstrap.sh` file:

* Uncommented the `az login --allow-no-subscriptions` command.

Removal of the `bootstrap/tenant-configuration/.terraform.lock.hcl` file:

* Removed the lock file, which is maintained automatically by "terraform init".

Additions to the `quickstarts/301-sap-gateway` directory:

* Added a new `README.md` file that provides a detailed guide for setting up a Terraform script to provision a Virtual Machine and all the requirements to connect the Self-hosted Integration Runtime, the Microsoft Gateway, and SAP .NET Connector.
* Added a new `README.md.tmpl` file that provides a template for the `README.md` file.
* Modified the `main.tf` file in the `gateway-principal` directory to include the required version of Terraform and the required providers, and to create a new Azure AD application and service principal for the gateway.
* Added a new `outputs.tf` file in the `gateway-principal` directory to output the client ID and secret of the service principal.
* Added a new `variables.tf` file in the `gateway-principal` directory to declare variables for the client ID, secret, and tenant ID of the service principal.Implementing basic Checkov checks for SAP quickstart.

This deployment is a public one, so requires public access for Storage Accounts and Key Vault.

I skip a couple of checks because is a public deployment and because a lack on permisions to implement RBAC and encription to Storage account.

I tested and deployed update by update each commit, can be deployed with no errors with the correct "local.tfvars" file.

### CODE REVIEW

**Comment by eduardodfmex** [comment]
@furlan can you help me to review this PR..?
Thank you in advance.

File: README.md [hunk 1]
```diff
@@ -13,8 +13,7 @@ This repository contains scripts quickly build out a new tenant and configure it
 
 ### Tooling
 
-> [!NOTE]
-> The following tooling is pre-installed in the [VS Code Dev Container](https://code.visualstudio.com/docs/devcontainers/containers) and it is highly recommended that you use the Dev Container to run the scripts and terraform modules in this repository:
+**_NOTE:_** The following tooling is pre-installed in the [VS Code Dev Container](https://code.visualstudio.com/docs/devcontainers/containers) and it is highly recommended that you use the Dev Container to run the scripts and terraform modules in this repository:
```
Comment (mawasile) [inline README.md:16]: if we don't generate docs from those document and just keep them in github page, then follow this markdowns:
https://github.com/orgs/community/discussions/16925

File: quickstarts/301-sap-gateway/gateway-principal/main.tf [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      source = "hashicorp/azurerm"
+    }
+    azuread = {
+      source = "hashicorp/azuread"
+    }
+  }
+}
+
+provider "azuread" {
+  client_id     = var.client_id
```
Comment (mawasile) [inline quickstarts/301-sap-gateway/gateway-principal/main.tf:14]: I would keep all examples using CLI unless there is a specific need of using client/secret

File: README.md (hunk 1 above)
Comment (eduardodfmex) [reply README.md:16]: Thank you for the comment, I try it, but doesn't work in the preview, look:

image

Comment (eduardodfmex) [reply README.md:16]: Thank you for the comment, I try it, but doesn't work in the preview, take a look:

image

**Review by furlan — DISMISSED** [review]
I reviewed the files and also tested the SAP connection through the SHIR created.
Everything looks good for me.

**Comment by furlan** [comment]
I reviewed the files and also tested the SAP connection through the SHIR created.
Everything looks good for me.

**Review by furlan — DISMISSED** [review]
I reviewed the files and also tested the SAP connection through the SHIR created.
Everything looks good for me.

**Review by furlan — APPROVED** [review]
LGTM!

**Pull request merged**